chainer/training/extensions/_snapshot.py

```diff
@@ -1,9 +1,96 @@
+import os
+
+import chainer
 from chainer.serializers import npz
 from chainer.training import extension
 from chainer.training.extensions import snapshot_writers
 from chainer.utils import argument
 
 
+def _find_snapshot_files(fmt, path):
+    '''Only prefix and suffix match
+
+    TODO(kuenishi): currently clean format string such as
+    "snapshot{.iteration}.npz" can only be parsed, but tricky (or
+    invalid) formats like "snapshot{{.iteration}}.npz" are hard to
+    detect and to properly show errors, just ignored or fails so far.
+
+    Args:
+        fmt (str): format string to match with file names of
+            existing snapshots, where prefix and suffix are
+            only examined. Also, files' staleness is judged
+            by timestamps. The default is metime.
+        path (str): a directory path to search for snapshot files.
+
+    Returns:
+        A sorted list of pair of ``mtime, filename``, whose file
+        name that matched the format ``fmt`` directoly under ``path``.
```

Comment:
A nitpick 
```suggestion
        name that matched the format ``fmt`` directly under ``path``.
```

Reply:
I included the fix on this in my commit.